lib/win32/certstore/store_base.rb

```diff
@@ -135,6 +135,20 @@ def read_certificate_content(cert_path)
         end
         File.read("#{cert_path}")
       end
+
+      def find_certificate(store_handler, certificate_name)
+        issuer_rdn_name = FFI::MemoryPointer.new(2, 256)
+        while (pCertContext = CertEnumCertificatesInStore(store_handler, pCertContext) and not pCertContext.null?)do
+          CertGetNameStringW(pCertContext, CERT_NAME_RDN_TYPE, CERT_NAME_ISSUER_FLAG, nil, issuer_rdn_name, 5000)
```

Comment:
the `5000` here and the `FFI::MemoryPointer` you're creating are risky.

[CertGetNameString](https://msdn.microsoft.com/en-us/library/windows/desktop/aa376086(v=vs.85).aspx) returns the size if the last argument, cchNameString, is zero. Then you can use that size to make a correctly sized `FFI::MemoryPointer` object and call `CertGetNameString` again.
I can't find an example pattern off hand, but there are other places in the code where we handle this.

Reply:
here are some examples: https://github.com/chef/chef/blob/master/lib/chef/win32/security.rb#L310